app/databse.py

```diff
@@ -0,0 +1,152 @@
+ID_CREATOR = get_id_creator()
+
+# DB field enums
+DOG_TYPE_ENUMS = Enum('lost', 'found', name='dog_type')
+DOG_SEX_ENUMS = Enum('male', 'female', name='dog_sex')
+
+# Create a base class for declarative models
+Base = declarative_base()
+
+
+class Dog(Base):
+    __tablename__ = 'dogs'
+
+    # Dog data
+    ## Entry data
+    id = Column(ID_COLUMN_TYPE, primary_key=True, default=ID_CREATOR)
+    dog_type = Column(DOG_TYPE_ENUMS, nullable=False)
+    is_matched = Column(Boolean, default=False)
+    is_verified = Column(Boolean, default=False)
+
+    ## contact details
+    contact_name = Column(String)
+    contact_phone = Column(String)
+    contact_email = Column(String)
+    contact_address = Column(String)
+
+    ## dog attributes
+    name = Column(String)
+    chip_number = Column(String(length=15))
+    breed = Column(String)
+    color = Column(String)
+    size = Column(String)
+    extra_details = Column(String)
+    sex = Column(DOG_SEX_ENUMS)
+
+    ## change times
+    created_at = Column(DateTime, default=datetime.utcnow)
+    updated_at = Column(DateTime, onupdate=datetime.utcnow)
+
+    # relationship
+    images = relationship('DogImages', back_populates='dog')
+
+
+class DogImages(Base):
+    __tablename__ = 'dog_images'
+    id = Column(ID_COLUMN_TYPE, primary_key=True, default=ID_CREATOR)
+    dog_id = Column(ID_COLUMN_TYPE, ForeignKey('dogs.id'))
+    image_url = Column(String)
+    created_at = Column(DateTime, default=datetime.utcnow)
+    updated_at = Column(DateTime, onupdate=datetime.utcnow)
+
+    dog = relationship('Dog', back_populates='images')
+
+
+def add_examples(session):
+    # Example 1: Insert a dog
+    dog1 = Dog(dog_type="lost", name="Rex", chip_number="1234567890", breed="Labrador",
+               color="Yellow", size="Large")
+    session.add(dog1)
+
+    # Example 2: Insert dog images
+    image1 = DogImages(dog=dog1, image_url="https://example.com/rex1.jpg")
+    image2 = DogImages(dog=dog1, image_url="https://example.com/rex2.jpg")
+    session.add_all([image1, image2])
+
+    # Commit the changes to the database
+    session.commit()
+
+    dog2 = Dog(dog_type="found", name="Buddy", chip_number="9876543210", breed="Golden Retriever",
+               color="Golden", size="Medium")
+    session.add(dog2)
+
+    # Commit the changes to the database
+    session.commit()
+
+    # Example 3: Insert more dog images
+    image3 = DogImages(dog=dog2, image_url="https://example.com/buddy1.jpg")
+    image4 = DogImages(dog=dog2, image_url="https://example.com/buddy2.jpg")
+    session.add_all([image3, image4])
+
+    # Commit the changes to the database
+    session.commit()
+
+
+def get_connection_string(user: str, password: str, host: str, port: int = 5432, db: str = 'dogfinder',
+                          protocol: str = 'postgresql') -> str:
+    if protocol == 'sqlite':
+        return f"sqlite:///{db}.db"
+    else:
+        return f'{protocol}://{user}:{password}@{host}:{port}/{db}'
+
+
+def main():
```

Comment:
This method needs to be commented out?

Reply:
This is creating the connection string:
- The "if" clause runs when there is not DB data in the environment variables, so the protocol defaults to "sqlite".
- if the environment variables contain real DB data the "else" clause runs.

Reply:
I meant the main method

Reply:
Wasn't sure about this. Will comment out.
Thanks.